pkg/bff/auth/authtest/server.go

```diff
@@ -34,6 +44,38 @@ const (
 	Scope        = "openid profile email"
 )
 
+var (
+	srv       *Server
+	srvErr    error
+	srvCreate sync.Once
+	srvClose  sync.Once
+)
+
+// Serve creates the singleton authtest server if it does not already exist and returns
+// it for use in tests and test dependency injection. If creating the server resulted in
+// an error then the error is returned. Once Close is called, this method will return
+// nil since the server is a singleton and can only be created once. Ensure that Close
+// is not called until the tests are complete.
+func Serve() (*Server, error) {
+	srvCreate.Do(func() {
+		srv, srvErr = New()
+	})
+	return srv, srvErr
+}
+
+// Close shuts down the single authtest server and cleans it up. This method should only
+// be called once when tests are completed. When the singleton server is shutdown it can
+// no longer be created a second time because of the use of sync.Once.
+func Close() {
+	srvClose.Do(func() {
+		if srv != nil {
+			srv.Close()
+		}
+		srv = nil
+		srvErr = errors.New("the authtest server has been closed")
+	})
+}
```

Comment:
Singleton module structure prevents us from having to figure out how to inject the authtest server client into the middleware.